pkg/distributor/otel_test.go

```diff
@@ -347,14 +378,53 @@ func TestHandlerOTLPPush(t *testing.T) {
 			},
 			responseCode: http.StatusOK,
 		},
+		{
+			name:        "Write histograms. With lz4 compression",
+			compression: "lz4",
+			maxMsgSize:  100000,
+			series: []prompb.TimeSeries{
+				{
+					Labels: []prompb.Label{
+						{Name: "__name__", Value: "foo"},
+					},
+					Histograms: []prompb.Histogram{
+						prompb.FromIntHistogram(1337, test.GenerateTestHistogram(1)),
+					},
+				},
+			},
+			metadata: []mimirpb.MetricMetadata{
+				{
+					Help: "metric_help",
+					Unit: "metric_unit",
+				},
+			},
+			verifyFunc: func(t *testing.T, pushReq *Request) error {
+				request, err := pushReq.WriteRequest()
+				require.NoError(t, err)
+
+				series := request.Timeseries
+				require.Len(t, series, 1)
+
+				histograms := series[0].Histograms
+				assert.Equal(t, 1, len(histograms))
+				assert.Equal(t, 1, int(histograms[0].Schema))
+
+				metadata := request.Metadata
+				assert.Equal(t, mimirpb.HISTOGRAM, metadata[0].GetType())
+				assert.Equal(t, "foo", metadata[0].GetMetricFamilyName())
+				assert.Equal(t, "metric_help", metadata[0].GetHelp())
+				assert.Equal(t, "metric_unit", metadata[0].GetUnit())
+
+				pushReq.CleanUp()
+				return nil
+			},
+			responseCode: http.StatusOK,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			exportReq := TimeseriesToOTLPRequest(tt.series, tt.metadata)
 			req := createOTLPProtoRequest(t, exportReq, tt.compression)
-			if tt.encoding != "" {
```

Comment:
I decided to remove this because we are doing this inside the `createOTLPProtoRequest` function.